packages/overlay/src/components/OpenInEditor.tsx

```diff
@@ -1,19 +1,21 @@
 import type React from 'react';
 import { useCallback } from 'react';
+import { useSpotlightContext } from '~/lib/useSpotlightContext';
 import { ReactComponent as PenIcon } from '../assets/pen.svg';
 
 export default function OpenInEditor({ file }: { file: string }) {
+  const { sidecarUrl } = useSpotlightContext();
+  const sidecarOpenUrl: string = new URL('/open', sidecarUrl).href;
```

Comment:
Since you abstracted out `sidecarUrl` and this `new URL(x, sidecarUrl).href` looks like a pattern, do you think we can change the variable from the context to a function:

```ts
const { getSidecarUrl } = useSpotlightContext();
const openUrl = getSidecarUrl('open');
```